packages/xo-web/src/common/xo/index.js

```diff
@@ -1313,7 +1314,37 @@ export const toggleMaintenanceMode = async host => {
       return
     }
   }
-  return _call('host.setMaintenanceMode', { id: resolveId(host), maintenance: host.enabled })
+  return _call('host.setMaintenanceMode', { id: resolveId(host), maintenance: host.enabled }).catch(async err => {
```

Comment:
I think we should check if there are VMs to force migrate before calling `host.setMaintenanceMode`, so it avoids an unnecessary call and it avoids making a comparison on an error message.

Reply:
I did it the same way as for sr.enableMaintenanceMode (a call, and another call in a catch), as asked in the ticket:

https://github.com/vatesfr/xen-orchestra/blob/f236238c73f2a72747a110ba1e47fbda42c78d7c/packages/xo-web/src/common/xo/index.js#L2730

Handling the problem on both the backend and frontend side can be seen as a duplication.

Reply:
Ok, then it looks good to me